mpas_analysis/ocean/climatology_map_waves.py

```diff
@@ -0,0 +1,528 @@
+#
+# Copyright (c) 2020 Triad National Security, LLC. All rights reserved.
+# Copyright (c) 2020 Lawrence Livermore National Security, LLC. All rights
+# reserved.
+# Copyright (c) 2020 UT-Battelle, LLC. All rights reserved.
+#
+# Additional copyright and license information can be found in the LICENSE file
+# distributed with this code, or at
+# https://raw.githubusercontent.com/MPAS-Dev/MPAS-Analysis/master/LICENSE
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import os
+import csv
+import xarray as xr
+import dask
+from multiprocessing.pool import ThreadPool
+from pyremap import ProjectionGridDescriptor
+
+from mpas_analysis.shared import AnalysisTask
+
+from mpas_analysis.shared.io.utility import build_obs_path, decode_strings, \
+    build_config_full_path
+from mpas_analysis.shared.io import write_netcdf
+
+from mpas_analysis.shared.climatology import RemapMpasClimatologySubtask, \
+    RemapObservedClimatologySubtask, get_antarctic_stereographic_projection
+from mpas_analysis.shared.climatology.climatology import \
+    get_masked_mpas_climatology_file_name
+
+from mpas_analysis.ocean.plot_climatology_map_subtask import \
+    PlotClimatologyMapSubtask
+
+from mpas_analysis.shared.constants import constants
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class ClimatologyMapWaves(AnalysisTask):  # {{{
+    """
+    An analysis task for the computation of wave climatologies
+    """
+    # Authors
+    # -------
+    # Steven Brus
+    # Xylar Asay-Davis
+
+    def __init__(self, config, mpasClimatologyTask, regionMasksTask,
+                 controlConfig):  # {{{
+        """
+        Construct the analysis task.
+
+        Parameters
+        ----------
+        config :  ``MpasAnalysisConfigParser``
+            Configuration options
+
+        mpasClimatologyTask : ``MpasClimatologyTask``
+            The task that produced the climatology to be remapped and plotted
+
+        regionMasksTask : ``ComputeRegionMasks``
+            A task for computing region masks
+
+        controlConfig :  ``MpasAnalysisConfigParser``
+            Configuration options for a control run
+        """
+        # Authors
+        # -------
+        # Steven Brus
+        # Xylar Asay-Davis
+
+        fields = [{'prefix': 'significantWaveHeight',
+                   'mpas': 'timeMonthly_avg_significantWaveHeight',
+                   'units': r'm',
+                   'titleName': 'Significant Wave Height'},
+                  {'prefix': 'peakWaveFrequency',
+                   'mpas': 'timeMonthly_avg_peakWaveFrequency',
+                   'units': r's',
+                   'titleName': 'Peak Wave Period'},
+                  #{'prefix': 'peakWavePeriod',
+                  # 'mpas': 'timeMonthly_avg_peakWavePeriod',
+                  # 'units': r's',
+                  # 'titleName': 'Peak Wave Period'},
+                   ]
```

Comment:
It seem like part of the problem is that you are using `prefix`' to be the name of the field in the `waveOutput` stream but it would be better to have that be the `mpas` name and to prepend `timeMonthly_avg` to it when you need to below:
```suggestion
        fields = [{'prefix': 'significantWaveHeight',
                   'mpas': 'significantWaveHeight',
                   'units': r'm',
                   'titleName': 'Significant Wave Height'},
                  {'prefix': 'peakWavePeriod',
                   'mpas': 'peakWaveFrequency',
                   'units': r's',
                   'titleName': 'Peak Wave Period'}]
```